source/sundials/kinsol.cc

```diff
@@ -148,6 +151,147 @@ namespace SUNDIALS
 
       return err;
     }
+
+    template <class VectorType>
+    int
+    t_kinsol_jacobian_vmult(
+      N_Vector     v,     /* vector that must be multiplied by the jacobian */
+      N_Vector     Jv,    /* where to store the result of the vmult */
+      N_Vector     u,     /* current value of the dependent variable vector */
+      booleantype *new_u, /* u has changed since last call to this function */
+      void *       user_data)
+    {
+      KINSOL<VectorType> &solver =
+        *static_cast<KINSOL<VectorType> *>(user_data);
+      GrowingVectorMemory<VectorType> mem;
+
+      typename VectorMemory<VectorType>::Pointer src_v(mem);
+      solver.reinit_vector(*src_v);
+
+      typename VectorMemory<VectorType>::Pointer src_u(mem);
+      solver.reinit_vector(*src_u);
+
+      typename VectorMemory<VectorType>::Pointer dst_jv(mem);
+      solver.reinit_vector(*dst_jv);
+
+      copy(*src_v, v);
+      copy(*src_u, u);
+
+      solver.jacobian_vmult(*src_v, *src_u, *dst_jv);
+
+      copy(Jv, *dst_jv);
+      *new_u = SUNFALSE;
```

Comment:
In older versions `SUNFALSE` is `FALSE`. Have a look at https://github.com/dealii/dealii/blob/ed62762abdcb72b2d427608c0b39ff29471bd090/source/sundials/arkode.cc#L143-L147.